src/store/modules/keycodes/index.js

```diff
@@ -27,17 +28,55 @@ const state = {
   active: 'ANSI'
 };
 
-function isISO() {
-  return store.state.app.configuratorSettings.iso;
+function getOSKeyboardLayout() {
+  const fallbackOSKeyboardLayout = 'keymap_us';
+  return (
+    store.state.app.configuratorSettings.osKeyboardLayout ||
+    fallbackOSKeyboardLayout
+  );
 }
 
-function generateKeycodes(isIso, isSteno) {
+function isANSI() {
+  return keymapExtras[getOSKeyboardLayout()].isANSI;
+}
+
+function toLocaleKeycode(keycodeLUT, keycodeObject) {
+  console.assert(!isUndefined(keycodeLUT));
+  if (
+    !Object.keys(keycodeObject).includes('name') ||
+    !Object.keys(keycodeObject).includes('code')
+  ) {
+    // Not an object describing a keyboard key; return as is
+    return keycodeObject;
+  }
+  if (keycodeLUT[keycodeObject.code]) {
+    // Clone in a shallow manner the original keycodeObject object and
+    // override the name, title, and possibly other fields
+    return { ...keycodeObject, ...keycodeLUT[keycodeObject.code] };
+  } else {
+    return keycodeObject;
+  }
+}
+
+function generateKeycodes(osKeyboardLayout, isSteno) {
+  store.state.app.configuratorSettings.iso = !isANSI();
   const keycodes = [
-    ...(isIso ? keycodeLayout.ISO : keycodeLayout.ANSI),
-    ...keycodeLayout.normal,
+    ...(isANSI() ? keycodePickerTabLayout.ANSI : keycodePickerTabLayout.ISO),
+    ...keycodePickerTabLayout.special,
     ...(isSteno ? steno : [])
   ];
-  return keycodes;
+  if (!Object.keys(keymapExtras).includes(getOSKeyboardLayout())) {
+    console.log(
+      `${getOSKeyboardLayout()} is not a valid OS keyboard layout value!`
+    );
+    return keycodes;
+  }
+  return keycodes.map((keycodeObject) =>
+    toLocaleKeycode(
+      keymapExtras[getOSKeyboardLayout()].keycodeLUT,
```

Comment:
hoist this line out of map. You're calling a function and then de-referencing the same data every invocation of map. That shouldn't change ever while you are inside the loop.

Reply:
Solved in https://github.com/qmk/qmk_configurator/pull/1161/commits/bbd6dc9c3b4361b52932eceb776990340280d1ce